src/throttler.guard.ts

```diff
@@ -80,7 +81,7 @@ export class ThrottlerGuard implements CanActivate {
     // Throw an error when the user reached their limit.
     if (ttls.length >= limit) {
       res.header('Retry-After', nearestExpiryTime);
-      this.throwThrottlingException(context);
+      this.throwThrottlingException(context, { retryAfter: nearestExpiryTime });
```

Comment:
Is there anything else we should pass here for more flexibility down the road, or is this the only metadata we want to add?

Reply:
From the top of my head, I don't see any more info that would be relevant for the error response. 

We could, however, also somehow pass this info to the controller method for the non-throttled case. So we can, for example, send throttling limits as part of the successful response body. Somethig like `@ThrottlingInfo()` parameter decorator, maybe?

In this latter case, I'd also add the request limit, and the remaining number of requests - that is, all the info that is currently added to the headers.

Reply:
That sounds like a good idea. We could attach it directly to the `ExecutionContext` so that it could be easily retrievable regardless the transport strategy via the custom decorator